src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -46,6 +47,10 @@ public function validate($value, Constraint $constraint)
         $variables['value'] = $value;
         $variables['this'] = $this->context->getObject();
 
+        if (strlen($constraint->dataPath) > 0) {
+            $variables['this'] = $this->getPropertyAccessor()->getValue($this->context, $constraint->dataPath);
```

Comment:
I like the idea, to pass the `root` value directly to `this` if it's null. No `dataPath` and so easier to use.